core/src/miner/miner.rs

```diff
@@ -837,13 +840,17 @@ impl MinerService for Miner {
         let parent_block_number = chain.block_header(&parent_block).expect("Parent is always exist").number();
         if self.requires_reseal(parent_block_number) {
             let (block, original_work_hash) = match self.prepare_block(parent_block, chain) {
-                Ok((block, original_work_hash)) => {
+                Ok(Some((block, original_work_hash))) => {
```

Comment:
BTW, `block` and `original_work_hash` is duplicated 3 times. It can be reduced with
```rust
let (block, original_work_hash) = match ... {
  Ok(Some(prepared @ (block, _))) => { 
    ...
    prepared
  }
  ...
};
```